pkg/api/types.go

```diff
@@ -306,6 +306,14 @@ type KubernetesConfig struct {
 	CtrlMgrNodeMonitorGracePeriod    string            `json:"ctrlMgrNodeMonitorGracePeriod,omitempty"`
 	CtrlMgrPodEvictionTimeout        string            `json:"ctrlMgrPodEvictionTimeout,omitempty"`
 	CtrlMgrRouteReconciliationPeriod string            `json:"ctrlMgrRouteReconciliationPeriod,omitempty"`
+	CustomFiles                      *[]CustomFile     `json:"customFiles,omitempty"`
```

Comment:
What is the reasoning behind putting this field in `KubernetesConfig`? In the example you gave it makes sense but could this feature be used to upload files to a master node for more general purposes? Does it only work for Kubernetes clusters?

Reply:
Good point. The feature itself can provision whatever files to whatever directory (where there's permission) so yeah, it's not conceptually a kubernetes feature. The current state of the PR only provisions files to master nodes. So either add as a field to `masterProfile` => `$.properties.masterProfile.customFiles or simply have it on its own, so `$.properties.masterCustomFiles` (in the kubernetes.json). What do you think?

Reply:
I like the former approach, let's move everything over into the `masterProfile` struct. Thanks!